optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/statistic/StatisticRegistry.java

```diff
@@ -92,7 +111,7 @@ public void extractScoreFromMeters(SolverMetric metric, Tags runId, Consumer<Sco
         for (int i = 0; i < labelNames.length; i++) {
             Gauge scoreLevelGauge = this.find(metric.getMeterId() + "." + labelNames[i]).tags(runId).gauge();
             if (scoreLevelGauge != null) {
-                levelNumbers[i] = scoreLevelGauge.value();
+                levelNumbers[i] = scoreLevelNumberConverter.apply(scoreLevelGauge.value());
```

Comment:
I'm thinking... shouldn't you be able to just build the score from all its levels, and then retrieve the level numbers from it?

Reply:
Score definitions expect the score levels to be a particular type (Exactly integer, exactly long, etc.). Micrometer gauges can only return doubles. So in order to not get a typecast exception in the score definition, I need to convert it to the "expected" type. (There are micrometer gauges for each level)